policy/modules/system/systemd.te

```diff
@@ -1417,3 +1417,516 @@ optional_policy(`
+allow systemd_homed_t self:netlink_kobject_uevent_socket create_socket_perms;
+
+
+
+#
+# dbus
+#
+
+# AVC avc:  denied  { acquire_svc } for  scontext=system_u:system_r:systemd_homed_t:s0 tcontext=system_u:system_r:system_dbusd_t:s0-s0:c0.c1023 tclass=dbus permissive=1 "
+optional_policy(`
+	dbus_acquire_svc_system_dbusd(systemd_homed_t)
+')
+
+# AVC avc:  denied  { send_msg } for  scontext=system_u:system_r:systemd_sleep_t:s0 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=dbus permissive=0
+allow systemd_sleep_t systemd_homed_t:dbus send_msg;
+
+#### Allow: Reboot/login
+# AVC avc:  denied  { send_msg } for  scontext=system_u:system_r:sshd_t:s0-s0:c0.c1023 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=dbus permissive=0
+# AVC avc:  denied  { connectto } for  pid=1000 comm="sshd" path="/run/systemd/userdb/io.systemd.Home" scontext=system_u:system_r:sshd_t:s0-s0:c0.c1023 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=unix_stream_socket permissive=0
+optional_policy(`
+	gen_require(`
+		type sshd_t;
+	')
+	allow sshd_t systemd_homed_t:dbus send_msg;
+	systemd_homed_userdb_stream_connect(sshd_t)
+')
+
+# AVC avc:  denied  { send_msg } for  scontext=system_u:system_r:xdm_t:s0-s0:c0.c1023 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=dbus permissive=0 exe="/usr/bin/dbus-broker" sauid=81 hostname=? addr=? terminal=?
+# AVC avc:  denied  { connectto } for  pid=1000 comm="gdm-session-wor" path="/run/systemd/userdb/io.systemd.Home" scontext=system_u:system_r:xdm_t:s0-s0:c0.c1023 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=unix_stream_socket permissive=0
+# AVC avc:  denied  { write } for  pid=1000 comm="gdm-session-wor" path="/run/systemd/home/testuser.dont-suspend" dev="tmpfs" ino=3614 scontext=system_u:system_r:xdm_t:s0-s0:c0.c1023 tcontext=system_u:object_r:systemd_homed_runtime_t:s0 tclass=fifo_file permissive=0
+optional_policy(`
+	gen_require(`
+		type xdm_t;
+	')
+	xserver_dbus_chat_xdm(systemd_homed_t)
+	systemd_homed_userdb_stream_connect(xdm_t)
+	allow xdm_t systemd_homed_runtime_t:fifo_file write;
+')
+
+# AVC avc:  denied  { send_msg } for  scontext=system_u:system_r:local_login_t:s0-s0:c0.c1023 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=dbus permissive=0
+# AVC avc:  denied  { connectto } for  pid=1000 comm="login" path="/run/systemd/userdb/io.systemd.Home" scontext=system_u:system_r:local_login_t:s0-s0:c0.c1023 tcontext=system_u:system_r:systemd_homed_t:s0 tclass=unix_stream_socket permissive=0
+# AVC avc:  denied  { write } for  pid=1000 comm="login" path="/run/systemd/home/testuser.dont-suspend" dev="tmpfs" ino=4784 scontext=system_u:system_r:local_login_t:s0-s0:c0.c1023 tcontext=system_u:object_r:systemd_homed_runtime_t:s0 tclass=fifo_file permissive=0
+optional_policy(`
+	gen_require(`
+		type local_login_t;
+	')
+	allow local_login_t systemd_homed_t:dbus send_msg;
+	systemd_homed_userdb_stream_connect(local_login_t)
+        allow local_login_t systemd_homed_runtime_t:fifo_file write;
+')
+
```

Comment:
Yes so this should probably be addressed on a PAM level (ie these rules probably apply to all login programs) Also probably best to not reference local_login_t directly here.